monday/client.py

```diff
@@ -7,26 +7,42 @@
 """
 
 from .__version__ import __version__
-from .resources import ItemResource, UpdateResource, TagResource, BoardResource, UserResource, GroupResource, ComplexityResource, WorkspaceResource, NotificationResource
+from .resources import (
+    CustomResource,
+    ItemResource,
+    ColumnsResource,
+    UpdateResource,
+    TagResource,
+    BoardResource,
+    UserResource,
+    GroupResource,
+    ComplexityResource,
+    WorkspaceResource,
+    NotificationResource,
+    MeResource
+)
 
 
 class MondayClient:
-    def __init__(self, token):
+    def __init__(self, token, headers={}):
```

Comment:
[non-blocking] Given that this header is there primarily to handle the API change, are we planning on getting rid of this header parameter after Monday removes the old API?

Reply:
They probably will still update the API, and still let us set the API version on the headers, so at least I think it would be helpful to have the option to set the headers to every request nonetheless.

We depend on them to make the changes, so we'll never know.

Reply:
Also, looks like this is the only commented code that I actually made, so this is the only one I can say about.

Reply:
Defaulting a parameter to a mutable value is very dangerous:

```python
client = MondayClient(token = "abc")
client2 = MondayClient(token = "MMM")

client.boards.fetch_columns_by_board_id("dsadsaa")
client2.boards.client.headers # {'Authorization': 'abc', 'Content-Type': 'application/json'}  wrong API key
```

Reply:
An alternative would be to pass the desired version `api_version="2023-10"` to the constructor

Happy to hear your thoughts but here's why I think it'd be better:

1. It makes it more obvious what we're trying to achieve with this param, since that's the only (current) logical use for the `headers`, outside of internal stuff like Content-Type etc.
2. By keeping the version on e.g. self.api_version, it would allow to keep the old stuff working if someone wants to use the old API for the time being. For example the added `items_page()` in queries could be added only if self.api_version == '2023-10' etc
3. See my comment above about mutating a default class param. A string doesn't have that issue
4. Users of the lib don't need to know what the headers look like (yes, it's very simple to document, but still)

Reply:
My $0.02:  I'm still of the opinion that we should consider burning this parameter entirely come 16 Jan 2024, and have version labels indicating what's happening.  

Given that Monday is dropping support for the old version on 15 Jan, this would be a legacy parameter 3 months after release.  There's certainly a chance that they'll do more versioning in the future, but I'd rather be handling it implicitly, through our client library versioning, than try to support multiple API versions in the same client library.  I appreciate that their future versioning might be more increment, but given that they're using date versioning, rather than semantic versioning, they're permitted hard breaks with each iteration.  

Also, this is how I wish we were doing it now, but I'm not one to look a gift PR in the mouth, given that we've got a relatively short timeline for cutting this library over.

Reply:
> My $0.02: I'm still of the opinion that we should consider burning this parameter entirely come 16 Jan 2024, and have library version labels indicating what's happening.
> 
> Given that Monday is dropping support for the old version on 15 Jan, this would be a legacy parameter 3 months after release. There's certainly a chance that they'll do more versioning in the future, but I'd rather be handling it implicitly, through our client library versioning, than try to support multiple API versions in the same client library. I appreciate that their future versioning might be more incremental, but given that they're using date versioning, rather than semantic versioning, they're permitted hard breaks with each iteration.
> 
> Also, this is how I wish we were doing it now, but I'm not one to look a gift PR in the mouth, given that we've got a relatively short timeline for cutting this library over.

I see your point, because they are literally deprecating the old API. Maybe there is simply no need for an extra parameter. In any case, it should definitely not be declared with a mutable dict as default params value :)